# GATK mutect2
# This pipeline is for running Mutect2 from GATK4.2.5, annotating and then subsetting to CGC genes of interest
# These are WGS data of tRCC pateints on 18 samples
# The first pipeline is variant calls using Mutect2 for a single sample run followed by filtering and annotation with SnpEff
# The run can be split into chromosomes or by specifying co-ordinates to run in parallel otherwise a single run can take ~60-90hrs!
# if scatter and gather method is used, additional steps of merging vcfs are required at the end

#!/bin/bash
source /etc/profile
#$ -S /bin/bash
#$ -pe pvm 16

#modules GATK, miniconda and picard loaded from what is already on Argos
#module used are as follows, comment out when necessary

module load /mnt/storage/spack/share/spack/modules/linux-ubuntu16.04-x86_64/gatk-4.2.5.0-gcc-5.4.0-tfxkdz3

# run mutect2 with gatk

gatk --java-options "-Xmx100G" Mutect2 \
-R /mnt/storage/labs/sviswanathan/GATK_ref/Homo_sapiens_assembly38.fasta \
-I /home/ma1111/tRCC_data/Meyerson_Srinivas_60XWGS_PCR_Nov2019/RP-2139/WGS/DTRCC3_Rkidney/v3/DTRCC3_Rkidney.cram \
-tumor DTRCC3_RKidney \
-I /home/ma1111/tRCC_data/Meyerson_Srinivas_30XWGS_Nov2019/RP-2139/WGS/DTRCC_3_Blood/v1/DTRCC_3_Blood.cram \
-normal DTRCC_3_Blood \
--germline-resource /mnt/storage/labs/sviswanathan/GATK_ref/af-only-gnomad.hg38.vcf.gz \
--panel-of-normals /mnt/storage/labs/sviswanathan/GATK_ref/1000g_pon.hg38.vcf.gz \
--f1r2-tar-gz /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/f1r2.tar.gz \
-O /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_f1r2_somatic4.2.vcf.gz

# for filtering-

# read oreintation filters, this can be run at the end too but before filtering!

gatk LearnReadOrientationModel 
-I /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/f1r2.tar.gz 
-O /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/read-orientation-model_DTRCC3.tar.gz \

# getpileup summaries for tumor 
gatk GetPileupSummaries \
-R /mnt/storage/labs/sviswanathan/GATK_ref/Homo_sapiens_assembly38.fasta \
-I /home/ma1111/tRCC_data/Meyerson_Srinivas_60XWGS_PCR_Nov2019/RP-2139/WGS/DTRCC3_Rkidney/v3/DTRCC3_Rkidney.cram \
-V /mnt/storage/labs/sviswanathan/GATK_ref/small_exac_common_3.hg38.vcf.gz \
-L /mnt/storage/labs/sviswanathan/GATK_ref/small_exac_common_3.hg38.vcf.gz \
-O /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_tum_pileup.table \

# getpileup summaries for normal
# uses matched normal
gatk GetPileupSummaries \
-R /mnt/storage/labs/sviswanathan/GATK_ref/Homo_sapiens_assembly38.fasta \
-I /home/ma1111/tRCC_data/Meyerson_Srinivas_30XWGS_Nov2019/RP-2139/WGS/DTRCC_3_Blood/v1/DTRCC_3_Blood.cram \
-V /mnt/storage/labs/sviswanathan/GATK_ref/small_exac_common_3.hg38.vcf.gz \
-L /mnt/storage/labs/sviswanathan/GATK_ref/small_exac_common_3.hg38.vcf.gz \
-O /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_norm_pileup.table \

# calculate contamination
gatk CalculateContamination \
-I /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_norm_pileup.table \
-matched /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_norm_pileup.table \
-O /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_matched_contamination_table \
-tumor-segmentation /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_tum_segments.table \

# filter mutect calls after adding segments table, tumor segmentation table while filtering is optional
# previously I didn't add tumor segmentation table while filtering, this tumor segmentation tabel is generated by calculate contamination
# when run with $ without tumor segmentation, with this table reduces no. of variants but its optional

gatk FilterMutectCalls \
-R /mnt/storage/labs/sviswanathan/GATK_ref/Homo_sapiens_assembly38.fasta \
-V /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_f1r2_somatic4.2.vcf.gz \
# -tumor-segmentation /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_tum_segments.table \
--contamination-table /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_matched_contamination.table \
--ob-priors /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/read-orientation-model_DTRCC3.tar.gz \
-O /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_filtered1_matched_somatic4.2.vcf.gz

# from this filtered vcf which is labelled as PASS/. select those that have passed the filter by using bcftoools
module load /mnt/storage/spack/share/spack/modules/linux-ubuntu16.04-x86_64/bcftools-1.9-gcc-5.4.0-ywzc5m3
bcftools view -i "%FILTER='PASS'" DTRCC3_filtered1_matched_somatic4.2.vcf.gz > DTRCC3_filtered1_passed_ann.vcf

# count the no. pass variants or analyze the stats

# annotate with Snpeff or Funcotator

# annotation run by snpeff with gatk

java -Xmx8g -jar /home/ma1111/tools/snpEff_v4_3t/snpEff.jar \ 
-c /home/ma1111/tools/snpEff_v4_3t/snpEff.config \ 
-v hg38 /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC7/ann/DTRCC7_filtered1_passed_seg_somatic4.2.vcf -o gatk \

> /home/ma1111/tRCC_data/mutect_run/mutect_run4.2.5.0/DTRCC3/DTRCC3_filtered1_passed_ann.vcf



